packages/fxa-payments-server/src/components/PlanDetails/index.tsx

```diff
@@ -260,56 +260,66 @@ export const PlanDetails = ({
                   </div>
                 )}
 
-                {exclusiveTaxRates.length === 1 && (
-                  <div className="plan-details-item">
-                    <Localized id="plan-details-tax">
-                      <div>Taxes and Fees</div>
-                    </Localized>
-
-                    <Localized
-                      id={`tax`}
-                      attrs={{ title: true }}
-                      vars={{
-                        amount: getLocalizedCurrency(
-                          exclusiveTaxRates[0].amount,
-                          currency
-                        ),
-                        intervalCount: interval_count,
-                      }}
-                    >
-                      <div data-testid="tax-amount">
-                        {getLocalizedCurrencyString(
-                          exclusiveTaxRates[0].amount,
-                          currency
-                        )}
-                      </div>
-                    </Localized>
-                  </div>
-                )}
-
-                {exclusiveTaxRates.length > 1 &&
-                  exclusiveTaxRates.map((taxRate, idx) => (
+                {exclusiveTaxRates.length === 1 &&
+                  exclusiveTaxRates[0].amount > 0 && (
                     <div className="plan-details-item">
-                      <div>{taxRate.display_name}</div>
+                      <Localized id="plan-details-tax">
+                        <div>Taxes and Fees</div>
+                      </Localized>
 
                       <Localized
                         id={`tax`}
-                        key={idx}
                         attrs={{ title: true }}
                         vars={{
                           amount: getLocalizedCurrency(
-                            taxRate.amount,
+                            exclusiveTaxRates[0].amount,
                             currency
                           ),
                           intervalCount: interval_count,
                         }}
                       >
-                        <div>
-                          {getLocalizedCurrencyString(taxRate.amount, currency)}
+                        <div data-testid="tax-amount">
+                          {getLocalizedCurrencyString(
+                            exclusiveTaxRates[0].amount,
+                            currency
+                          )}
                         </div>
                       </Localized>
                     </div>
-                  ))}
+                  )}
+
+                {exclusiveTaxRates.length > 1 &&
+                  exclusiveTaxRates.map((taxRate, idx) => {
+                    if (taxRate.amount === 0) {
+                      return null;
+                    } else {
```

Comment:
Instead of this, would you mind using a filter on line 95? It would eliminate all of the changes here and above.

Reply:
done